internal/uvm/scsi.go

```diff
@@ -313,7 +374,6 @@ func (uvm *UtilityVM) RemoveSCSI(ctx context.Context, hostPath string) error {
 func (uvm *UtilityVM) GetScsiUvmPath(ctx context.Context, hostPath string) (string, error) {
```

Comment:
This is only called once in the entire project and where its called does not have access to a scsimount. I could be misunderstanding though

Reply:
The thing I really don't like about functions like this is that the UVM is returning information that hcsoci should already know. Yes, in the place where GetScsiUvmPath is called, hcsoci does not have a ScsiMount today. But in hcsioci.allocateWindowsResources, we got the ScsiMount! Where did it go? Why do we have to ask uvm for such information again?

We should not have extra random "getters" in one module (uvm) because we couldn't figure out how to connect two code paths in a second module (hcsoci). Frankly, I suspect a better design for all of this would be for hcsoci to manage the ref counts and get all that junk out of uvm to begin with.

Reply:
(Note that this problem is preexisting so it doesn't need to be fixed now. But I think we should reconsider the split between uvm and hcsoci.)

Reply:
(Note that this problem is preexisting so it doesn't need to be fixed now. But I think we should reconsider the split between uvm and hcsoci.)

Reply:
Agreed. Hcsoci, and Resources follows suit as its a member, is so heavily tied to the uvm package right now as all of the Add/Remove/Modify<ResourceName> functions are all uvm methods. Makes it so you have to always have a handle to a uvm to do anything regarding resources. This would be the biggest change by far if these two packages merged or we could find a way to make them depend on each other less but i dont see how. I dont have the best solution in my head at the moment for this

Reply:
One way to think about this is: what if we wanted to make an interface (or multiple interfaces) for uvm and have a second implementation, e.g. for testing or to support a different virtualization stack? How much functionality would have to be duplicated into each uvm implementation today, and how could we move some of it to hcsoci so that the uvm package is only concerned with the specifics of VMs, not ref counting and so on.

Reply:
Ok that clears it up a little more, we can talk about this tomorrow in more detail